File: app/src/main/java/com/denchic45/kts/data/model/mapper/SectionMapper.java
package com.denchic45.kts.data.model.mapper;

import com.denchic45.kts.data.model.domain.Section;
import com.denchic45.kts.data.model.room.SectionEntity;

import org.mapstruct.Mapper;

@Mapper
public interface SectionMapper extends DomainEntityMapper<Section, SectionEntity> {
}

File: app/src/main/java/com/denchic45/kts/data/model/mapper/SectionMapper.kt
package com.denchic45.kts.data.model.mapper

import com.denchic45.kts.domain.model.Section
import com.denchic45.kts.data.remote.model.SectionMap
import com.denchic45.kts.data.model.room.SectionEntity
import org.mapstruct.Mapper
import org.mapstruct.Named

@Mapper
abstract class SectionMapper {

    @Named("entityToDomain")
    fun entityToDomain(entity: SectionEntity?): Section {
        return entity?.let {
            sectionEntityToSectionDomain(it)
        } ?: Section.createEmpty()
    }

    abstract fun sectionEntityToSectionDomain(entity: SectionEntity): Section

    abstract fun entityToDomain(entityList: List<SectionEntity>): List<Section>

    abstract fun docToEntity(sectionMap: SectionMap): SectionEntity

    abstract fun docToEntity(sectionDocs: List<SectionMap>): List<SectionEntity>
}